### DISCUSSION TITLE
bonsai-sleap pose tracking issue

### DISCUSSION THREAD

**Discussion by SSsriranga** [bonsai-rx/bonsai#1358] [Q&A] [answered]
I have trained a Single-Instance SLEAP model and have been trying to setup the Bonsai-SLEAP interface for real-time tracking. I used the PredictSinglePose workflow that I found on the Bonsai.SLEAP git repo and have successfully set it up but there are a few tracking issues that I could use your help with! 

The issue is that while the body parts are tracked, the positions are not mapped out correctly onto the video. For instance, in the screenshot attached, the FileCapture video panel shows the nose, ears (R and L) and tail positions (in green) incorrectly placed in the centre of the frame instead of on the rat itselt. Furthermore, the tracked positions are always restriced to the centre of the field of view and never move to the periphery suggesting that they are incorrectly projected onto the video. 

The videos used to label and train the SLEAP network have the same dimensions (1280 x 1024) as the testing video used in the workflow. 

I have attached a link with the zip folder containing all the necessary files to run the script -- any advice would be very helpful!

Thanks, 
Sonali 

![bonsai_sleap_workflow_snapshot](https://github.com/bonsai-rx/bonsai/assets/77621614/8f0cef27-fdf2-4723-9d80-869b542160f8)
ZIP file: https://www.dropbox.com/s/jzj4dbp7rwwh0fc/bonsai_sleap_rat_arena.zip?dl=0

**Comment by bruno-f-cruz** [comment]
Hey!
It seems to me that what you are seeing makes sense given that the input to the network is a cropped area around the centroid any not the full original frame. Am I missing something perhaps?

**Reply by SSsriranga** [reply]
Hi Bruno, 

Thanks for your prompt reply! I didn't intent to input the network with a cropped area. If you're referring to the CropCenter node, I have it down as the entire width/high of the video as removing it caused errors with the PredictSinglePost node!

**Comment by bruno-f-cruz** [comment]
Not sure I follow. CropCenter crops and area around the animal, as a result
the position will always be “centered” in the image. Can you describe what
would be the desired output you are looking for?

**Reply by SSsriranga** [reply]
The desired output is to have the body part positions along the same dimensions as the frames in video that I'm trying to track -- For instance, the screenshot above will have the green positions (in the FileCapture panel) over the rats ears, nose and tail and not centred to the middle. The reason I have CropCenter in this workflow is because it was on your github page and I wasn't able to run the script by disabling it and I assumed a workaround would be to input the size of the entire frame in CropCenter.

**Reply by bruno-f-cruz** [reply, marked as answer]
Is the input size error solved by [this](https://github.com/orgs/bonsai-rx/discussions/1312#discussioncomment-5477111) ?

**Reply by SSsriranga** [reply]
I tried adding ResizeCanvas to the current workflow and it resulted in an error: 
![workflow1](https://github.com/bonsai-rx/bonsai/assets/77621614/de079aa0-ca94-4d02-96be-22628136407f)


Simplifying the workflow produces the desired output but the video is playing at approx. 1.5x speed; I'm not sure if this will be a problem during real-time tracking or if its a video encryption issue? 
![workflow2](https://github.com/bonsai-rx/bonsai/assets/77621614/67145e72-c076-443e-8285-1da92d4b3ccd)

**Reply by bruno-f-cruz** [reply]
The first problem is to be expected, since you are inputting a pair of a centroid and an image and the operator has no idea what you want to do with it.

Regarding the second workflow, you are saying that the workflow runs faster than before? Is that not a good thing?

**Reply by SSsriranga** [reply]
I see! Sorry about this, I'm very new to Bonsai and still understanding the workflows. 
The video in the second workflow is playing at a much faster speed and I assumed that was an issue but I suppose you are right!

**Reply by bruno-f-cruz** [reply]
No worries, keep us posted on your progress!
cheers,
B

**Reply by SSsriranga** [reply]
Thank you for your help!!
